src/components/tableRow/TableRow.tsx

```diff
@@ -3,47 +3,60 @@ import './TableRow.scss';
 import Button from '../button/Button';
 import { FiEdit } from 'react-icons/fi';
 import { RiDeleteBin5Line } from 'react-icons/ri';
+import { returnClassName } from '../utils/tools';
+import { ReactNode } from 'react';
 
 interface IProps {
-  name?: string;
-  email?: string;
-  role?: number;
-  index?: number;
+  item?: any;
+  classPrefix?: string;
+  children?: ReactNode;
 }
 
 const TableRow = ({
-  name = 'john smith',
-  email = 'john@mail.com',
-  role = 1,
-  index = 0,
+  item = { name: 'testing', email: 'testing@mail.com', role: 100 },
+  classPrefix = '',
+  children,
 }: IProps) => {
+  const keys = Object.keys(item);
+  console.log(children);
+
   return (
     <tr className="tableRow">
-      <td className="index">{index}</td>
-      <td className="name">
-        <NameTag name={name} />
-      </td>
-      <td className="mail">{email}</td>
-      <td className="role">{role}</td>
+      {/* if there's a children it will return children wrapped in <td> */}
```

Comment:
TableRow could be better handled if you render the row by the item data, don't provide children or if you need to as a second option you can pass it as is and let the user put what ever he wants as children
But the most convenient will be rendering over the Item data and you may provide a renderCell prop to give user option to render what ever he needs over the cell itself.